src/hostname/hostnamed.c

```diff
@@ -1265,9 +1291,20 @@ static int method_set_static_hostname(sd_bus_message *m, void *userdata, sd_bus_
         if (streq_ptr(name, c->data[PROP_STATIC_HOSTNAME]))
                 return sd_bus_reply_method_return(m, NULL);
 
-        if (name && !hostname_is_valid(name, 0))
+        if (name && !hostname_is_valid(name, VALID_HOSTNAME_QUESTION_MARK))
```

Comment:
I'd like to move the validity check after substituting, like `method_set_hostname()` does.

BTW, method_set_hostname() and method_set_static_hostname() here do the same thing.
Maybe better to introduce a tiny helper function?